server/data/search.go

```diff
@@ -42,15 +42,14 @@ func SearchUsers(queryFields [][2]string) []*UserProfile {
 	rows, err := db.Query(context.Background(), sqlQuery, queryValues...)
 
 	if err != nil {
-		fmt.Println("Error searching for users in field. Error:", err.Error())
+		return []*UserProfile{}, err
 	}
 
 	usersFound, err := pgx.CollectRows(rows, pgx.RowToAddrOfStructByNameLax[UserProfile])
 
 	if err != nil {
-		fmt.Println("Error collecting rows in SearchUsers. Error:", err.Error())
-		return []*UserProfile{}
+		return []*UserProfile{}, err
```

Comment:
same here. I'd also like to point out that typically empty slices are created using the `make` function and slice literals are used when making a slice with a small number of initial values inside. It's just a common convention though, so I am nitpicking here.